src/compiler/scala/tools/nsc/ast/parser/Scanners.scala

```diff
@@ -103,11 +124,11 @@ trait Scanners extends ScannersCommon {
     }
   }
 
-  abstract class Scanner extends CharArrayReader with TokenData with ScannerData with ScannerCommon {
+  abstract class Scanner extends CharArrayReader with TokenData with ScannerData with ScannerCommon with DocScanner {
     private def isDigit(c: Char) = java.lang.Character isDigit c
 
     private var openComments = 0
-    protected def putCommentChar(): Unit = nextChar()
+    final protected def putCommentChar(): Unit = { processCommentChar(); nextChar() }
```

Comment:
We don't make those sorts of firm compatibility guarantees for compiler internals. 

In the past, I think our shared backstop for this sort of thing was `scripts/jobs/integrate/ide` — if we broke something the IDE was using, it would tell us. But that script was never updated to work with 2.12, so it hasn't been part of PR validation or nightly testing for a long time now.

If we want to avoid this kind of thing happening, I think either that script would need to be restored to be part of Scala's again, or (to catch the problem a little later, but still relatively fast), you would have to add something comparable to your own CI that tests your code against 2.12.x (and 2.13.x?) nightlies.